cum/scrapers/mangadex.py

```diff
@@ -0,0 +1,143 @@
+    pages_re = re.compile(r'var page_array ?= ?\[([^\]]+)\]', re.DOTALL)
+    # Just extract the single page name like: x1.jpg
+    single_page_re = re.compile(r'\s?\'([^\']+)\',?')
+    # This can be a mirror server or data path. Example:
+    # var server = 'https://s2.mangadex.com/'
+    # var server = '/data/'
+    server_re = re.compile(r'var server ?= ?\'([^\']+)\'')
+    uses_pages = True
+
+    @staticmethod
+    def _reader_get(url, page_index):
+        return requests.get(url)
+
+    def available(self):
+        self.r = self.reader_get(1)
+        if not len(self.r.text):
+            return False
+        elif self.r.status_code == 404:
+            return False
+        elif re.search(re.compile(r'Chapter #[0-9]+ does not exist.'),
+                       self.r.text):
+            return False
+        else:
+            return True
+
+    def download(self):
+        if getattr(self, 'r', None):
+            r = self.r
+        else:
+            r = self.reader_get(1)
+        soup = BeautifulSoup(r.text, config.get().html_parser)
+        chapter_hash = re.search(self.hash_re, r.text).group(1)
+        pages_var = re.search(self.pages_re, r.text)
+        pages = re.findall(self.single_page_re, pages_var.group(1))
+        files = [None] * len(pages)
+        mirror = re.search(self.server_re, r.text).group(1)
+        server = 'https://mangadex.com' + mirror if mirror[0] == '/' else mirror
+        futures = []
+        last_image = None
+        with self.progress_bar(pages) as bar:
+            for i, page in enumerate(pages):
+                if guess_type(page)[0]:
+                    image = server + chapter_hash + '/' + page
+                else:
+                    print('Unkown image type for url {}'.format(page))
+                    raise ValueError
+                r = requests.get(image, stream=True)
+                if r.status_code == 404:
+                    r.close()
+                    raise ValueError
+                fut = download_pool.submit(self.page_download_task, i, r)
+                fut.add_done_callback(partial(self.page_download_finish,
+                                              bar, files))
+                futures.append(fut)
+                last_image = image
+            concurrent.futures.wait(futures)
+            self.create_zip(files)
+
+    def from_url(url):
+        r = MangadexChapter._reader_get(url, 1)
+        soup = BeautifulSoup(r.text, config.get().html_parser)
+        try:
+            series_url = soup.find('a', href=MangadexSeries.url_re)['href']
+        except TypeError:
+            raise exceptions.ScrapingError('Chapter has no parent series link')
+        series = MangadexSeries('https://mangadex.com' + series_url
+                                if series_url[0] == '/'
+                                else series_url)
+        for chapter in series.chapters:
+            if chapter.url.lstrip('htps') == url.lstrip('htps'):
```

Comment:
lstrip seems like a bad way to remove the protocol part, if that's the intention here. `url.split('://', maxsplit=1)[1]` isn't very pretty but should strip any sort of protocol without going beyond the protocol part of an url.

Reply:
Or maybe `''.join(urllib.parse.urlparse(url)[1:3])` to deal with URLs with no scheme, but that's ugly too. I dunno